package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +777,635 @@ def __setstate__(self, state):
+        HDF5 dataset compression setting to be applied
+        to position, velocity, and force datasets. Allowed
+        settings are 'gzip', 'szip', 'lzf'. If an integer
+        in range(10), this indicates gzip compression level.
+        Otherwise, an integer indicates the number of a
+        dynamically loaded compression filter.
+    compression_opts : int or tup (optional)
+        Compression settings.  This is an integer for gzip, 2-tuple for
+        szip, etc. If specifying a dynamically loaded compression filter
+        number, this must be a tuple of values. For gzip, 1 indicates
+        the lowest level of compression and 9 indicates maximum compression.
+    positions : bool (optional)
+        Write positions into the trajectory [``True``]
+    velocities : bool (optional)
+        Write velocities into the trajectory [``True``]
+    forces : bool (optional)
+        Write forces into the trajectory [``True``]
+    timeunit : str (optional)
+        option to convert values in the 'time' dataset to a custom unit,
+        must be recognizable by MDAnalysis
+    lengthunit : str (optional)
+        option to convert values in the 'position/value' dataset to a
+        custom unit, must be recognizable by MDAnalysis
+    velocityunit : str (optional)
+        option to convert values in the 'velocity/value' dataset to a
+        custom unit, must be recognizable by MDAnalysis
+    forceunit : str (optional)
+        option to convert values in the 'force/value' dataset to a
+        custom unit, must be recognizable by MDAnalysis
+    author : str (optional)
+        name of the author of the file
+    author_email : str (optional)
+        email of the author of the file
+    creator : str (optional)
+        software that wrote the file [``MDAnalysis``]
+    creator_version : str (optional)
+        version of software that wrote the file
+        [:attr:`MDAnalysis.__version__`]
+
+
+    Note
+    ----
+    Parellel writing with the use of a MPI communicator and the ``'mpio'``
+    HDF5 driver is currently not supported.
+
+
+    Raises
+    ------
+    RuntimeError
+        when `H5PY`_ is not installed
+    ValueError
+        when ``n_atoms`` is 0
+    ValueError
+        when the file is opened with ``driver=='mpio'`` or
+        with an MPI communicator
+    ValueError
+        when ``chunks=False`` but the user did not specify ``n_frames``
+    ValueError
+        when ``positions``, ``velocities``, and ``forces`` are all
+        set to ``False``
+    TypeError
+        when the input object is not a :class:`Universe` or
+        :class:`AtomGroup`
+    IOError
+        when ``n_atoms`` of the :class:`Universe` or :class:`AtomGroup`
+        being written does not match ``n_atoms`` passed as an argument
+        to the writer
+    ValueError
+        when any of the optional ``timeunit``, ``lengthunit``,
+        ``velocityunit``, or ``forceunit`` keyword arguments are
+        unrecognizable by MDAnalysis
+
```

Comment:
Put the long text section here. See numpy doc.

Reply:
You have a lot of great docs hidden in private methods. Many describe limitations or caveats with the format or writer. The notes section should include this information, too.